### DISCUSSION TITLE
How to reduce 429 throttling using a personal account

### DISCUSSION THREAD

**Discussion by jamesapyrich** [abraunegg/onedrive#2827] [Q&A] [answered]
Hello! I love the fact this tool exists; I've been looking for a OneDrive->Linux solution for a while but my focus was always on scripting w/ OneDrive directly... but I can script against files super easily, so this is excellent. Plus, I now have a local backup of the drive on Linux :)

Anyway! I'm doing an initial sync, and the folder of interest is a folder shared to my personal account. It contains at most a few thousand files, but quite a lot of data (685GB). I'm getting some serious ramps in `Retry-After` when getting rate limited with the Microsoft Graph API (it'll start at something like 300s, and then jump to 10221 seconds or more), so I was wondering what I can do from my side to reduce the Graph API calls frequency, or to make them appear less "costly" to Microsoft so I have more wiggle room.

I see that this question has been asked before and I do understand it's a Microsoft backend issue... but hopefully I can slow my client's roll and get penalized less, because what I'm seeing is pretty harsh.

It occurred to me as I write this that I do have OneDrive set up to sync to at least 2 other personal computers, but I've disabled downloading by default as I don't want to spare the space on those systems.

This particular shared folder is (I believe) shared from another personal account, and there is a SaaS app that will write to that folder, but writing is fairly infrequent (2-3 times a day, 6-8 files at a time).

I am running `onedrive` on an Alpine box, and I'm using podman running rootless using the `latest` tag, which I pulled last night.

Thanks, in advance!

James

**Comment by jamesapyrich** [comment]
Here's my current config file:
```
sync_dir = "/onedrive/data"
download_only = "true"
sync_dir_permissions = "755"
sync_file_permissions = "644"
cleanup_local_files = "false"
local_first = "false"
monitor_interval = "3600"
```

I only care about getting remote changes, I will never upload anything from here, and if things get deleted remotely, I want to keep them.

**Comment by abraunegg** [comment, marked as answer]
> How to reduce 429 throttling using a personal account?

1. Use the correct application version on your platform - v2.5.0 or greater. As you are using Docker, ensure you are using `edge` not `latest` to ensure you get fixes. Please read the Docker guide on this: https://github.com/abraunegg/onedrive/blob/master/docs/docker.md
2. Ensure you have installed Docker the *correct* way - that is - from the Docker packages, not your Linux distribution - this is important.
3. Adjust your 'threads' downwards - https://github.com/abraunegg/onedrive/blob/master/docs/application-config-options.md#threads - default is 8, you may need to decrease to 2
4. Increase your 'monitor_interval' to 3600 (1 hour) or to 'match' your online data rate of change - https://github.com/abraunegg/onedrive/blob/master/docs/application-config-options.md#monitor_interval
5. Potentially upgrade from a Personal Account to a Business Account

**Reply by jamesapyrich** [reply]
Makes sense, thanks for your prompt response! I am using `podman` as `docker` install (albeit via the package manager) led to me throwing my hands in the air multiple times. But, I've otherwise followed the `podman` procedure as indicated. I'll consider upgrading to Business if I get rate limited with the above suggestions.
